addons/mass_mailing/static/src/js/mass_mailing_snippets.js

```diff
@@ -151,4 +242,215 @@ options.registry.ImageTools.include({
+     * @override
+     */
+    async start() {
+        const res = await this._super(...arguments);
+        const $editable = this.options.wysiwyg.getEditable();
+        this.styleElement = $editable[0].ownerDocument.querySelector('#design-element');
+        if (!this.styleElement) {
+            // If a style element can't be found, create one and initialize it.
+            this.styleElement = document.createElement('style');
+            this.styleElement.setAttribute('id', 'design-element');
+            // The style element needs to be within the layout of the email in
+            // order to be saved along with it.
+            $editable.find('.o_layout').prepend(this.styleElement);
+            this.styleElement.textContent = DEFAULT_CSS;
+        }
+        // When editing a stylesheet, its content is not updated so it won't be
+        // saved along with the mailing. Therefore we need to write its cssText
+        // into it. However, when doing that we lose its reference. So we need
+        // two separate style elements: one that will be saved and one to hold
+        // the stylesheet. Both need to be synchronized, which will be done via
+        // `_commitCss`.
+        let sheetOwner = $editable[0].ownerDocument.querySelector('#sheet-owner');
+        if (!sheetOwner) {
+            sheetOwner = document.createElement('style');
+            sheetOwner.setAttribute('id', 'sheet-owner');
+            $editable[0].ownerDocument.head.appendChild(sheetOwner);
+        }
+        sheetOwner.disabled = true;
+        sheetOwner.textContent = this.styleElement.textContent;
+        this.styleSheet = sheetOwner.sheet;
+        return res;
+    },
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Option method to set a css property in the mailing's custom stylesheet.
+     * Note: marks all the styles as important to make sure they take precedence
+     * on other stylesheets.
+     *
+     * @param {boolean|string} previewMode
+     * @param {string} widgetValue
+     * @param {Object} params
+     * @param {string} params.selectorText the css selector for which to apply
+     *                                     the css
+     * @param {string} params.cssProperty the name of the property to edit
+     *                                    (camel cased)
+     * @param {string} [params.toggle] if 'true', will remove the property if
+     *                                 its value is already the one it's being
+     *                                 set to
+     * @param {string} [params.activeValue] the value to set, if `widgetValue`
+     *                                      is not defined.
+     * @returns {Promise|undefined}
+     */
+    customizeCssProperty(previewMode, widgetValue, params) {
+        if (!params.selectorText || !params.cssProperty) {
+            return;
```

Comment:
I'm curious, when can this happen in practice ?

Reply:
IIRC, when resetting after a preview.